From f03a1769c55b4992613e153bd8d7bd012fc67525 Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Fri, 5 Dec 2014 18:17:39 +0100
Subject: [PATCH] core: keep the password only if the user part is set

- a report by Timo Klecker

(cherry picked from commit 8c3d167b63096d3b465c53aca92ee18c5255188c)
---
 action.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/action.c b/action.c
index 2e58034..43cac50 100644
--- a/action.c
+++ b/action.c
@@ -932,8 +932,8 @@ int do_action(struct run_act_ctx* h, struct action* a, struct sip_msg* msg)
 
 				if (a->type==SET_USERPASS_T) tmp=0;
 				else tmp=uri.passwd.s;
-				/* passwd */
-				if (tmp){
+				/* passwd - keep it only if user is set */
+				if (user && tmp){
 					len=uri.passwd.len; if(crt+len+1>end) goto error_uri;
 					*crt=':'; crt++;
 					memcpy(crt,tmp,len);crt+=len;
-- 
2.1.3

